src/data/colleges/en.ts

```diff
@@ -111,37 +129,51 @@ const engineeringRequirements: readonly CollegeOrMajorRequirement[] = [
       'https://www.engineering.cornell.edu/students/undergraduate-students/advising/liberal-studies',
     checker: [
       (course: Course): boolean =>
-        engineeringLiberalArtsDistributions.some(
-          distribution => hasCategory(course, distribution) ?? false
+        Object.values(engineeringLiberalArtsGroups).some(group =>
+          group.some(distribution => hasCategory(course, distribution) ?? false)
         ) || courseIsForeignLang(course),
     ],
     fulfilledBy: 'courses',
     perSlotMinCount: [6],
     slotNames: ['Course'],
     additionalRequirements: {
-      'Courses must be from 3 categories.': {
+      'Courses must be from 3 groups.': {
         checker: [
           (course: Course): boolean =>
-            (course.catalogDistr?.includes('LA') || course.catalogDistr?.includes('LAD')) ?? false,
-          ...engineeringLiberalArtsDistributions
-            .filter(it => it !== 'LA' && it !== 'LAD')
-            .map(distribution => (course: Course): boolean =>
-              hasCategory(course, distribution) ?? false
+            engineeringLiberalArtsGroups['Group 1'].some(
+              distribution => hasCategory(course, distribution) ?? false
+            ),
+          (course: Course): boolean =>
+            engineeringLiberalArtsGroups['Group 2'].some(
+              distribution => hasCategory(course, distribution) ?? false
+            ),
+          (course: Course): boolean =>
+            engineeringLiberalArtsGroups['Group 3'].some(
+              distribution => hasCategory(course, distribution) ?? false
+            ),
+          (course: Course): boolean =>
+            engineeringLiberalArtsGroups['Group 4'].some(
+              distribution => hasCategory(course, distribution) ?? false
+            ),
+          (course: Course): boolean =>
+            engineeringLiberalArtsGroups['Group 5'].some(
+              distribution => hasCategory(course, distribution) ?? false
+            ),
+          (course: Course): boolean =>
+            engineeringLiberalArtsGroups['Group 6'].some(
+              distribution => hasCategory(course, distribution) ?? false
             ),
         ],
         fulfilledBy: 'courses',
-        perSlotMinCount: [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1],
-        slotNames: [
-          'LA',
-          ...engineeringLiberalArtsDistributions.filter(it => it !== 'LA' && it !== 'LAD'),
-        ],
+        perSlotMinCount: [1, 1, 1, 1, 1, 1],
+        slotNames: Object.keys(engineeringLiberalArtsGroups),
         minNumberOfSlots: 3,
       },
       'Courses must have at least 18 credits.': {
         checker: [
           (course: Course): boolean =>
-            engineeringLiberalArtsDistributions.some(
-              distribution => hasCategory(course, distribution) ?? false
+            Object.values(engineeringLiberalArtsGroups).some(group =>
```

Comment:
Same comment as before, but it's just nitpicking for the sake of clean code to be honest, you don't have to implement.